### DISCUSSION TITLE
How to import PostgresClient type?

### DISCUSSION THREAD

**Discussion by TheGP** [weldsorm/welds#41] [Q&A]
from welds connections
```
#[derive(Debug, Clone)]
pub struct PostgresClient {
    pool: Arc<PgPool>,
}
```

**Comment by TheGP** [comment]
I think its
```
use welds::connections::postgres::PostgresClient;
```
You should add the type to example, connections are often made as static

**Comment by lex148** [comment]
Hey @TheGP you are correct that is the concrete type for a postgres connection. I typically do not recommend using it directly however.  

I would recommend using functions that look like this where you can get away with it.

```rust
use welds::Client;
async fn do_sql_work(client: &dyn Client) {
    client.execute("select 1", &[]).await.unwrap();
}
```

This has the advantage that it could be any flavor of SQL and also could be a transaction if needed.

I do realize that sometimes you need a concrete type too. If that is the situation that you are in, you might want to try this.
In the `main.rs` or `lib.rs` make a type wrapper around the specific connection type you are using.

_main.rs_
```rust
type DbClient = welds::connection::postgres::PostgresClient;

async fn do_sql_work(client: &DbClient) {
    client.execute("select 1", &[]).await.unwrap();
}

async fn do_more_sql_work(client: &DbClient) {
    client.execute("select 2", &[]).await.unwrap();
}
```

and use your type instead of the specific one in welds.
This has the advantage that if you need to switch from postgres to say sqlite for testing it is a one line change.


Hope this helps, If not lets chat some more. I want to help to get what you need.

@lex148

**Comment by TheGP** [comment]
@lex148 Got it, thank you!
